modular_robot_physical/revolve2/modular_robot_physical/_physical_robot_controller.py

```diff
@@ -0,0 +1,154 @@
+            parser.add_argument(
+                "physical_robot_config"  # TODO: add type once set
+            )  # os.fsencode is bytes  mybe think of other way
+            parser.add_argument("--hardware", type=str)
+            parser.add_argument(
+                "--debug", help="Print debug information", action="store_true"
+            )
+            parser.add_argument(
+                "--dry",
+                help="If set, gpio output is skipped.",
+                action="store_true",
+            )
+            parser.add_argument(
+                "--log", help="If set, outputs controller log to this file.", type=str
+            )
+            parser.add_argument(
+                "--all",
+                help="Set all outputs provided in the config file to the given value.",
+                type=str,
+                choices=["min", "center", "max"],
+            )
+            args = parser.parse_args()
+            self._log_file = args.log
+            self._log = []
+
+            self._config = PhysicalRobotConfig.from_pickle(args.physical_robot_config)
+            self._control_period = 1 / self._config.control_frequency
+
+            self._sensor_interface = PhysicalSensorState()
+            match args.hardware:
+                case "v1":
+                    self._hardware_interface = V1PhysicalControlInterface(
+                        dry=args.dry,
+                        debug=args.debug,
+                        hinge_mapping=self._config.hinge_mapping,
+                        inverse_pin=self._config.inverse_servos,
+                    )
+                case "pca9685":
+                    self._hardware_interface = Pca9685PhysicalControlInterface(
+                        dry=args.dry,
+                        debug=args.debug,
+                        hinge_mapping=self._config.hinge_mapping,
+                        inverse_pin=self._config.inverse_servos,
+                    )
+                case _:
+                    raise NotImplementedError(
+                        "There is no physical interface for this type of hardware."
+                    )
+
+            match args.all:
+                case "min":
+                    self._hardware_interface.set_servo_targets(
+                        [-1.0 for _ in self._config.hinge_mapping]
+                    )
+                    input("Press enter to stop.\n")
+                case "max":
+                    self._hardware_interface.set_servo_targets(
+                        [1.0 for _ in self._config.hinge_mapping]
+                    )
+                    input("Press enter to stop.\n")
+                case "center":
+                    self._hardware_interface.set_servo_targets(
+                        [0.0 for _ in self._config.hinge_mapping]
+                    )
+                    input("Press enter to stop.\n")
+                case _:
+                    self._hardware_interface.careful = True
```

Comment:
what is careful?

Reply:
not moving all servos instantly to avoid hardware failure
so after a servo moved it waits for x seconds to move the next